networkx/readwrite/nx_ete.py

```diff
@@ -0,0 +1,215 @@
+"""
+****
+ETE
+****
+Read and write NetworkX graphs in the New Hampshire Newick format.
+
+"ETE (Environment for Tree Exploration) is a Python programming toolkit
+that assists in the automated manipulation,
+analysis and visualization of phylogenetic trees.
+It uses the New Hampshire Newick format as one of the most widely used
+standard representation of trees in bioinformatics."
+See http://etetoolkit.org/ for documentation.
+
+Format
+------
+http://etetoolkit.org/docs/latest/tutorial/tutorial_trees.html#reading-and-writing-newick-trees
+
+"""
+
+__all__ = ["to_ete", "from_ete", "write_ete", "read_ete"]
+
+import networkx as nx
+from networkx.utils import open_file, deque
+
+
+def to_ete(G, root_node=None):
+    """Returns a ETE Tree from NetworkX Graph G.
+
+    ETE (Environment for Tree Exploration) is a Python programming toolkit
+    that assists in the automated manipulation,
+    analysis and visualization of phylogenetic trees [2]_.
+    It uses the New Hampshire Newick format as one of the most widely used
+    standard representation of trees in bioinformatics [1]_.
+
+
+    Parameters
+    ----------
+    G : graph
+       A NetworkX graph.
+    root_node : node
+       A root of the tree.
+
+    Returns
+    -------
+    T : ETE Tree
+
+    Examples
+    --------
+    >>> G = nx.path_graph(42)
+    >>> T = nx.to_ete(G)
+
+    References
+    ----------
+    .. [1] http://etetoolkit.org/docs/latest/tutorial/tutorial_trees.html#reading-and-writing-newick-trees
+    .. [2] http://etetoolkit.org/
+    """
+    assert nx.is_tree(G), "write_ete() requires `G` to be a tree: %s" % (
+        "https://networkx.org/documentation/stable/reference/algorithms/tree.html"
+    )
+
+    try:
+        import ete3
+    except ImportError as e:
+        raise ImportError("to_ete() requires ete3: http://etetoolkit.org/") from e
+
+    if root_node is None:
+        root_node = list(G.nodes)[0]
+
+    G_tree = nx.dfs_tree(G, source=root_node)
```

Comment:
Can this cause a reordering/restructuring of the input tree? If so, users should be aware of it. Alternatively, perhaps an implementation could be found that doesn't require this preprocessing. I'm not sure though.